aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp

```diff
@@ -46,26 +46,29 @@ static void addcdiv_cpu_kernel(TensorIterator& iter, Scalar value) {
   });
 }
 
-static void smooth_l1_backward_cpu_kernel(TensorIterator& iter, Scalar norm) {
+static void smooth_l1_backward_cpu_kernel(TensorIterator& iter, Scalar norm, double beta) {
   ScalarType dtype = iter.dtype(0);
   AT_DISPATCH_ALL_TYPES(dtype, "smooth_l1_backward_cpu_out", [&] {
     auto norm_val = norm.to<scalar_t>();
+    scalar_t beta_val(beta);
     auto norm_val_vec = Vec256<scalar_t>(norm_val);
+    auto beta_val_vec = Vec256<scalar_t>(beta_val);
     const auto neg_1_vec = Vec256<scalar_t>(-1);
     const auto pos_1_vec = Vec256<scalar_t>(1);
     cpu_kernel_vec(iter,
       [=](scalar_t input, scalar_t target, scalar_t grad_output) -> scalar_t {
         const auto x = input - target;
-        if (x < -1.)
+        if (x < -beta)
           return -norm_val * grad_output;
-        else if (x > 1.)
+        else if (x > beta)
           return norm_val * grad_output;
         else
-          return norm_val * x * grad_output;
+          return norm_val * x * grad_output / beta;
```

Comment:
looks like if `x == beta == 0`, we could reach here, and then trigger division by 0?

Reply:
I might miss sth, but looks like this is different from the match given in the doc. 

```
0.5 (x_i - y_i)^2 / beta, & \text{if } |x_i - y_i| < beta 
```

Should the three condition be `x <= -beta`, `x >= beta`, and otherwise?

Reply:
yep you're right, thanks. And using `x <= -beta` and `x >= beta` should fix the division by 0 issue.